File: Grunt/Grunt/Models/HaloInfinite/PlayerType.cs
// <copyright file="PlayerType.cs" company="Den Delimarsky">
// Developed by Den Delimarsky.
// Den Delimarsky licenses this file to you under the MIT license.
// See the LICENSE file in the project root for more information.
// The underlying API powering Grunt is managed by 343 Industries and Microsoft. This wrapper is not endorsed by 343 Industries or Microsoft.
// </copyright>

namespace OpenSpartan.Grunt.Models.HaloInfinite
{
    /// <summary>
    /// Type of player participating in a match.
    /// </summary>
    public enum PlayerType
    {
        /// <summary>
        /// Player is a human.
        /// </summary>
        Human = 1,

        /// <summary>
        /// Player is a bot.
        /// </summary>
        Bot = 2,
    }
}

File: src/node/src/models/halo-infinite/enums/player-type.ts
/**
 * Type of player in a match.
 */
export const PlayerType = {
  /** Human player */
  Human: 'Human',
  /** Bot/AI player */
  Bot: 'Bot',
} as const;

/**
 * Type representing valid player type values.
 */
export type PlayerType = (typeof PlayerType)[keyof typeof PlayerType];
